test/integration/bad/profiling/stanc.expected

```diff
@@ -0,0 +1,83 @@
+  $ ../../../../../install/default/bin/stanc profile-bad1.stan
+
+Syntax error in 'profile-bad1.stan', line 2, column 12 to column 13, parsing error:
+   -------------------------------------------------
+     1:  transformed data {
+     2:      profile {
+                     ^
+     3:  
+     4:      }
+   -------------------------------------------------
+
+Expected profile name in paranthesis.
+
+
+  $ ../../../../../install/default/bin/stanc profile-bad2.stan
+
+Syntax error in 'profile-bad2.stan', line 2, column 12 to column 13, parsing error:
+   -------------------------------------------------
+     1:  transformed data {
+     2:      profile() {
+                     ^
+     3:  
+     4:      }
+   -------------------------------------------------
+
+Expected profile name in paranthesis.
+
+
+  $ ../../../../../install/default/bin/stanc profile-bad3.stan
+
+Syntax error in 'profile-bad3.stan', line 2, column 13 to column 18, parsing error:
+   -------------------------------------------------
+     1:  transformed data {
+     2:      profile( while {
+                      ^
+     3:  
+     4:      }
+   -------------------------------------------------
+
+Expected profile name in paranthesis.
+
+
+  $ ../../../../../install/default/bin/stanc profile-bad4.stan
+
+Syntax error in 'profile-bad4.stan', line 2, column 13 to column 18, parsing error:
+   -------------------------------------------------
+     1:  transformed data {
+     2:      profile (while) {
+                      ^
+     3:  
+     4:      }
+   -------------------------------------------------
+
+Expected profile name in paranthesis.
+
+
+  $ ../../../../../install/default/bin/stanc profile-bad5.stan
+
+Syntax error in 'profile-bad5.stan', line 2, column 9 to column 16, parsing error:
+   -------------------------------------------------
+     1:  transformed data {
+     2:      real profile;
+                  ^
+     3:  }
+   -------------------------------------------------
+
+Identifier expected after type in top-level variable declaration.
```

Comment:
TBH I dont like this error message but dont know how to 

If this is some other keyword like
```stan
transformed data {
    real while;
}
```
its a more informative message.

Identifier 'while' clashes with reserved keyword.

There is probably no trivial way of keeping profile unreserved?

Reply:
Keyword tokens remain "unreserved" when listed in `identifier`
https://github.com/stan-dev/stanc3/blob/2c167b64a7eb9078500782ff8c988d562f329287/src/frontend/parser.mly#L132-L136
Although that may create ambiguity in the grammar.
You can change the error by listing the keyword under `decl_identifier` instead
https://github.com/stan-dev/stanc3/blob/2c167b64a7eb9078500782ff8c988d562f329287/src/frontend/parser.mly#L141-L148

Reply:
I think just a better error message would be fine, especially if unreserving it would cause ambiguity.

Reply:
Also thanks!